Editorial review: Document scroll-triggered animations (#43158)
mdn/content: files/en-us/web/css/reference/properties/timeline-trigger-activation-range/index.md

## Description

The `timeline-trigger-activation-range` property can be used to set a custom **activation range** for a [CSS scroll-triggered animation](/en-US/docs/Web/CSS/Guides/Animation_triggers/Using_scroll-triggered_animations) trigger. The activation range is the range between the scrolling offset at which the animated element's animation is activated, and the scrolling offset at which the animated element's animation is deactivated.

By default, the activation range along the timeline is `cover`, which means that the animation activates when the start edge of the trigger enters the end edge of the viewport, and deactivates when the end edge of the trigger has exited either edge of the viewport. This can be changed by setting a different `timeline-trigger-activation-range` value.

For example:

```css
.trigger {
  timeline-trigger-name: --my-trigger;
  timeline-trigger-source: view();
  timeline-trigger-activation-range: entry 0% exit 50%;
}
```

> An element with these declarations set will have an identifying {{cssxref("timeline-trigger-name")}} of `--my-trigger`, and a {{cssxref("timeline-trigger-source")}} value of `view()`, which selects the element's nearest ancestor scrolling element to define the timeline along which the activation range is measured.

Reviewer: ```suggestion
An element with these declarations set will have an identifying {{cssxref("timeline-trigger-name")}} of `--my-trigger`, and a {{cssxref("timeline-trigger-source")}} value of `view()`, which sets the source of the trigger to the view progress timeline tracking the element inside its nearest scrolling ancestor element along which the activation range is measured.
```

.

> We also set a `timeline-trigger-activation-range` value of `entry 100% exit 50%`, which sets the start of the activation range to 100% of the way through the entry range and the end of the activation range to 50% of the way through the `exit` range. The result is that the animated element's animation will activate when the trigger element has completely entered the viewport, and deactivate when 50% of the trigger element has left the viewport.

Reviewer: ```suggestion
We also set a `timeline-trigger-activation-range` value of `entry 100% exit 50%`, which sets the start of the activation range to 100% of the way through the entry range and the end of the activation range to 50% of the way through the `exit` range. The result is that the animated element's trigger will activate when the tracked element has completely entered the viewport, and deactivate when 50% of the tracked element has left the viewport.
```

> An animated element can be triggered by the previously-described trigger element by referencing the trigger's identifying name in its {{cssxref("animation-trigger")}} property.

Reviewer: ```suggestion
An animated element can be triggered by the previously-described trigger by referencing the trigger's identifying name in its {{cssxref("animation-trigger")}} property.
```

Author: updated

For example:

```css
.animated {
  animation: rotate 3s infinite linear both;
  animation-trigger: --my-trigger play-forwards play-backwards;
}
```

> In this case, the animation will be triggered by the trigger element with the `timeline-trigger-name` of `--my-trigger`. Its {{cssxref("animation-action")}} keywords — `play-forwards play-backwards` — specify that the animation should play forwards on activation, and backwards on deactivation.

Reviewer: Same remarks about:

- The animation is not getting triggered, but the trigger is. When that trigger becomes active (or inactive), the animation will do _something_.
- The use of the term “trigger element”

  suscipit lorem efficitur molestie tempus. Integer sit amet neque et risus
  iaculis sodales sed eget diam. Quisque sodales nunc sapien, vitae lacinia ex
  luctus quis. Maecenas scelerisque scelerisque elit eu consequat. Etiam ac
  tristique tellus, sed tincidunt velit.
</p>
```

#### CSS

The animated {{htmlelement("div")}} element has an `animation` applied that rotates it. We set an {{cssxref("animation-trigger")}} value on it that references a trigger name of `--t`; we also specify two {{cssxref("animation-action")}} values — `play` and `pause` — which specify that the animation will play on activation, and pause on deactivation.

The trigger `<div>` element is defined as the animated `<div>`'s trigger using a {{cssxref("timeline-trigger-name")}} value of `--t`, which is equal to the identifier referenced in the animated `<div>`'s `animation-trigger` property value, associating the two together. It also includes a {{cssxref("timeline-trigger-source")}} value of [`view()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/view), which sets the element providing the view progress timeline as the nearest scrolling ancestor element.

> Finally for this rule, we specify a `timeline-trigger-activation-range` of `entry 50% exit 50%`, which sets the trigger's activation range to a range between 50% through the `entry` range (when 50% of the trigger element has entered the viewport) and 50% through the `exit` range (when 50% of the trigger element has exited the viewport).

Reviewer: ```suggestion
Finally for this rule, we specify a `timeline-trigger-activation-range` of `entry 50% exit 50%`, which sets the trigger's activation range to a range between 50% through the `entry` range (when 50% of the trigger element has entered the viewport) and 50% through the `exit` range (when 50% of the tracked element has exited the viewport).
```
